backend/de.metas.business/src/main/java/de/metas/product/ProductRepository.java

```diff
@@ -280,6 +280,18 @@ public static BPartnerProduct ofBPartnerProductRecord(@NonNull final I_C_BPartne
 				.usedForVendor(record.isUsedForVendor())
 				.build();
 	}
+
+	public void resetCurrentVendorFor(@NonNull final ProductId productId)
+	{
+		queryBL.createQueryBuilder(I_C_BPartner_Product.class)
+				.addEqualsFilter(I_C_BPartner_Product.COLUMNNAME_M_Product_ID, productId)
+				.addOnlyActiveRecordsFilter()
```

Comment:
..also please don't update all `I_C_BPartner_Product`s  ut just the one that *needs* updating